core/src/main/java/greencity/service/impl/UserServiceImpl.java

```diff
@@ -715,6 +715,50 @@ public void addNewFriend(Long userId, Long friendId) {
         return userProfilePictureDtoList;
     }
 
+    /**
+     * Save user profile information {@link User}.
+     *
+     * @author Marian Datsko
+     */
+    @Override
+    public UserProfileDtoResponse saveUserProfile(UserProfileDtoRequest userProfileDtoRequest, MultipartFile image,
+                                                  String email) {
+        User user = userRepo
+            .findByEmail(email)
+            .orElseThrow(() -> new WrongEmailException(USER_NOT_FOUND_BY_EMAIL + email));
+        user.setFirstName(userProfileDtoRequest.getFirstName());
+        user.setCity(userProfileDtoRequest.getCity());
+        user.setUserCredo(userProfileDtoRequest.getUserCredo());
+        user.setSocialNetworks(userProfileDtoRequest.getSocialNetworks());
+        user.setShowLocation(userProfileDtoRequest.getShowLocation());
+        user.setShowEcoPlace(userProfileDtoRequest.getShowEcoPlace());
+        user.setShowShoppingList(userProfileDtoRequest.getShowShoppingList());
+        if (image != null) {
+            String url = fileService.upload(image).toString();
+            user.setProfilePicturePath(url);
+        }
+        try {
+            userRepo.save(user);
+        } catch (DataIntegrityViolationException e) {
+            throw new NotSavedException(ErrorMessage.USER_NOT_SAVED);
```

Comment:
By expecting this exception you are saying: I haven't implemented all the required validation rules yet. Do you plan to implement them in a separate pull request?